src/code/z_scene.c

```diff
@@ -279,59 +279,55 @@ void func_80098958(GlobalContext* globalCtx, SceneCmd* cmd) {
 }
 
 // Scene Command 0x0B: Object List
-#ifdef NON_MATCHING
-// this function still needs some work but it should be functionally equivalent
 void func_8009899C(GlobalContext* globalCtx, SceneCmd* cmd) {
     s32 i, j, k;
-    s16* objectEntry;
     ObjectStatus* status;
     ObjectStatus* status2;
-    ObjectStatus* status3;
+    s16* objectEntry;
     void* nextPtr;
 
     objectEntry = SEGMENTED_TO_VIRTUAL(cmd->objectList.segment);
-    status2 = &globalCtx->objectCtx.status[globalCtx->objectCtx.unk_09];
     k = 0;
-    status = status2;
     i = globalCtx->objectCtx.unk_09;
+    status = &globalCtx->objectCtx.status[i];
+    
     while (i < globalCtx->objectCtx.num) {
         if (status->id != *objectEntry) {
-            status3 = status2;
+            status2 = &globalCtx->objectCtx.status[i];
             for (j = i; j < globalCtx->objectCtx.num; j++) {
-                status3->id = 0;
-                status3++;
+                status2->id = 0;
+                status2++;
             }
             globalCtx->objectCtx.num = i;
             func_80031A28(globalCtx, &globalCtx->actorCtx);
-        } else {
-            i++;
-            status++;
-            k++;
-            objectEntry++;
-            status2++;
+            
+            continue;
         }
+        
+        i++;
+        k++;
+        objectEntry++;
+        status++;
     }
 
     if (cmd->objectList.num > OBJECT_EXCHANGE_BANK_MAX) {
+        if (1) {}
```

Comment:
i believe the only thing holding this up is that we wanted to try putting the if (1) {} outside of the assert condition. does that still match?